### ISSUE TITLE
“You do not have permission to edit this.” after update from 1.6.1.24 to 1.7.6.3

### ISSUE THREAD

**Issue by ballashop** [PrestaShop/PrestaShop#17471]
**Describe the bug**
Hi, with our ISP we updated from **1.6.1.24** to **1.7.6.3** in a sub folder so the site is not operative but it's quite to ready.
The data seems right.
I have a problem with permission.
When a try to install theme or modules we get this message:
“You do not have permission to edit this.”
I download all site and db to test it in local server, the same thing happens.
Thanks for any replies

**To Reproduce**

The ISP used classic update with the native module
I have the log.txt

**Screenshots**
“You do not have permission to edit this.”
the image show the warning in italian language.
<img width="813" alt="Schermata 2020-02-01 alle 12 30 50" src="https://user-images.githubusercontent.com/28685316/73591438-b783e100-44ee-11ea-81de-95507f2aff3c.png">

**Additional information**
PrestaShop version: 1.7.6.3
PHP version: 7.1

**Comment by khouloudbelguith** [comment]
Hi @ballashop,

I did not manage to reproduce the issue with PS1.7.6.3 after an upgrade from PS1.6.1.24 using the 1-click upgrade v4.10.1.
I attached a screen record
https://drive.google.com/file/d/1-hmJQiHk9fff9BR_CDkGAxltFyZJcz66/view
What is the exact version did you use during the upgrade?
Are you using an employee with a super admin profile?

Thanks!

**Comment by ballashop** [comment]
Hi @khouloudbelguith 
thanks a lot to reply me.
Can I send you the log.txt via email?
The ISP started from 1.6.1.24 to 1.7.6.3 by 1-click upgrade (it is likely that the version used is v4.10.1).
About a user yes, I'm using a super admin one.
Thanks a lot for your support.

**Comment by khouloudbelguith** [comment]
@ballashop,

Can you please provide us the log of your upgrade? It can be found in the log.txt file, available in `<admin folder>/autoupgrade/tmp/log.txt`.
We need also to retrieve the PHP error log and the debug mode report in order to find out what's wrong.
Don't you know how to get this information? Please read the following article:
http://build.prestashop.com/howtos/misc/how-to-create-bug-report/
This is a public space.
You can provide me the module by email.
My address mail: khouloud.belguith@prestashop.com
Let's be safe!

Thanks!

**Comment by ballashop** [comment]
dear @khouloudbelguith ,
I sent you a email with logs.
I'm waiting your kind response.
Thanks for your support.

**Comment by khouloudbelguith** [comment]
@ballashop, thanks!
You have some warnings during the upgrade.
1. >[WARNING] SQL 1.7.5.0
            1071 in ALTER TABLE `ps_customer`
  CHANGE `company` `company` VARCHAR(255),
  CHANGE `email` `email` VARCHAR(255) NOT NULL,
  CHANGE `passwd` `passwd` VARCHAR(255) NOT NULL: Specified key was too long; max key length is 1000 bytes</div>

2. >[WARNING] SQL 1.7.5.0
            1071 in ALTER TABLE `ps_employee`
  CHANGE `firstname` `firstname` VARCHAR(255) NOT NULL,
  CHANGE `email` `email` VARCHAR(255) NOT NULL,
  CHANGE `passwd` `passwd` VARCHAR(255) NOT NULL,
  CHANGE `lastname` `lastname` VARCHAR(255) NOT NULL: Specified key was too long; max key length is 1000 bytes</div>

3. >[WARNING] SQL 1.7.6.2
            1054 in INSERT INTO `ps_info_lang` SELECT
`entity`.`id_info`, 1, `entity`.`id_lang`, `entity`.`text`
FROM `ps_info_lang` entity
LEFT JOIN `ps_info_lang` entity2 ON `entity2`.`id_shop` = 1 AND `entity`.`id_info` = `entity2`.`id_info`
WHERE `entity2`.`id_shop` IS NULL: Unknown column 'entity2.id_shop' in 'where clause'</div>

4. >[WARNING] SQL 1.7.0.0
            1062 in INSERT INTO `ps_hook` (`name`, `title`, `description`, `position`) VALUES
  ('actionValidateCustomerAddressForm', 'Customer address form validation', 'This hook is called when a customer submit its address form', '1'),
  ('displayAfterCarrier', 'After carriers list', 'This hook is displayed after the carrier list in Front Office', '1'),
  ('displayCarrierExtraContent', 'Display additional content for a carrier (e.g pickup points)', 'This hook calls only the module related to the carrier, in order to add options when needed.', '1'),
  ('validateCustomerFormFields', 'Customer registration form validation', 'This hook is called to a module when it has sent additional fields with additionalCustomerFormFields', '1'),
  ('displayProductExtraContent', 'Display extra content on the product page', 'This hook expects ProductExtraContent instances, which will be properly displayed by the template on the product page.', '1'),
  ('displayNavFullWidth', 'Navigation', 'This hook displays full width navigation menu at the top of your pages', '1'),
  ('displayAfterBodyOpeningTag', 'Very top of pages', 'Use this hook for advertisement or modals you want to load first.', '1'),
  ('displayBeforeBodyClosingTag', 'Very bottom of pages', 'Use this hook for your modals or any content you want to load at the very end.', '1'): Duplicate entry 'displayAfterBodyOpeningTag' for key 'hook_name'</div>

Thanks to check those tables.
You can find the full schema for 1.7.6.3 here:
https://github.com/PrestaShop/PrestaShop/blob/1.7.6.3/install-dev/data/db_structure.sql

Thanks!

**Comment by ballashop** [comment]
Hi @khouloudbelguith,
thanks a lot,
Can I ask you the kindness to indicate which of these warnings concerns the administrator's permissions?
So I can send this message to the ISP to fix it.
Thanks a lot.

**Comment by ballashop** [comment]
Dear @khouloudbelguith 
I'm in contact with our ISP to fix the problem.
As I have useful information, I will write here.
Thanks a lot.

**Comment by khouloudbelguith** [comment]
@ballashop, Thanks!
Waiting for your feedback.

**Comment by ballashop** [comment]
Desar @khouloudbelguith 
the ISP fixed the problem, they wrote:

**"_the problem was generated by an update bug of version 1.6 to 1.7 in the access a authentication tables_"**

Unfortunately I don't have more specific information, I'm sorry.
Thanks a lot.

**Comment by khouloudbelguith** [comment]
@ballashop, To fix this issue you need to replace the `ps_access` table in DB with the one from clean PS1.7 installation.
You can, for example, install PrestaShop separately, copy & paste the table ps_access to your shop.
Or you can follow this solution in this comment to solve your issue https://github.com/PrestaShop/PrestaShop/issues/9513#issuecomment-415011961
Before any modification, you can make a backup of your old database.
Thanks to check & feedback

**Comment by ballashop** [comment]
Dear @khouloudbelguith 
really thanks for the message that really useful.
I've found another error about renew, the ISP wrote: the error it depend on a bug of Prestashop after update from 1.6 to 1.7.
Can I post here or I have to open another issue?
Thanks for you support.

**Comment by khouloudbelguith** [comment]
@ballashop, yes, you can post it here.

Thanks!

**Comment by ballashop** [comment]
Ok @khouloudbelguith 

when I click to the renew (dashboard)

![renw](https://user-images.githubusercontent.com/28685316/73747058-ddabba00-4756-11ea-8f5c-955a8affdaa7.gif)

I get this message from debug:

![renew-error](https://user-images.githubusercontent.com/28685316/73747081-e9977c00-4756-11ea-93a7-efb44e714531.png)

Thanks for your support.

**Comment by khouloudbelguith** [comment]
@ballashop, could you please try to re-install the `productcomments` module with removing the folder, clear cache, check & feedback.

Thanks!

**Comment by ballashop** [comment]
Hi @khouloudbelguith 
Ok, I'll try now, thanks.

**Comment by khouloudbelguith** [comment]
@ballashop, thanks!
Waiting for your feedback.

**Comment by ballashop** [comment]
Hi, @khouloudbelguith 
I didn't find this module :( into the module folder.
Maybe I'm wrong path.
I'm waiting your kind response.
Thanks.

**Comment by khouloudbelguith** [comment]
@ballashop, in your BO => Modules => Module manager page => if you search for the productcomment, you can't find it?
https://github.com/PrestaShop/productcomments
Thanks!

**Comment by ballashop** [comment]
@khouloudbelguith 
no unfortunately!
how can it be?
I looked for a clean install of 1.7.6.3 and there is no such module either.
Thanks

**Comment by khouloudbelguith** [comment]
@ballashop, yes!
This module is not automatically installed.
But the link "Product Reviews" in the dashboard is related to this module.
https://drive.google.com/file/d/1kzmuibruja4rhD1xorExUCMFhhSrMTSI/view
If the module is not installed, this link is not displayed.
![image](https://user-images.githubusercontent.com/16067358/73750699-e94eaf00-475d-11ea-91c6-cf2dedc83358.png)

Thanks!

**Comment by ballashop** [comment]
Hi @khouloudbelguith 
is it possible that from 1.6.1.24 to 1.7.6.3 (update by native PS module) the dashboard show the box also without the module?
for some reason he brought the data back but not the module.
I have the renew link in dashboard:

<img width="630" alt="Schermata 2020-02-04 alle 12 25 31" src="https://user-images.githubusercontent.com/28685316/73750976-72fe7c80-475e-11ea-81eb-affdf90f662c.png">

That link points to a module that doesn't exist, why the error appear, maybe!
The ISP rename module 1.6 folder, 

![module-rename](https://user-images.githubusercontent.com/28685316/73751652-cf15d080-475f-11ea-8655-85f5325270aa.png)

Can I take the module folder, the link you wrote before:
https://github.com/PrestaShop/productcomments
and copy it to the server, within module folder?
Thanks for your support.

**Comment by khouloudbelguith** [comment]
@ballashop, the `productcomments` module is compatible with PS1.7.6.0 or more: https://github.com/PrestaShop/productcomments/blob/master/productcomments.php#L60
It is not compatible with PS1.6.1.24
![image](https://user-images.githubusercontent.com/16067358/73752541-6f202980-4761-11ea-90c3-f0f3cbe07654.png)

Thanks!

**Comment by ballashop** [comment]
@khouloudbelguith 
ok, of course :)
the clean site is version 1.7.6.3, i can test first on that.
Obviously I intended to operate on 1.7
I'll let you know.
Thanks a lot.

**Comment by khouloudbelguith** [comment]
@ballashop, thanks!
Waiting for your feedback.

**Comment by ballashop** [comment]
hi @khouloudbelguith 
I tried to install the module I downloaded from:
https://github.com/PrestaShop/productcomments

<img width="286" alt="Schermata 2020-02-04 alle 16 00 15" src="https://user-images.githubusercontent.com/28685316/73756187-721e1880-4767-11ea-9e53-232c56a35fce.png">

but it didn't work:

<img width="685" alt="Schermata 2020-02-04 alle 15 59 40" src="https://user-images.githubusercontent.com/28685316/73756254-895d0600-4767-11ea-9beb-3ecf732ec9fe.png">

Do you have any other suggestions on how to proceed?
Thanks for your support.

**Comment by ballashop** [comment]
@khouloudbelguith 
i tried do install by module catalog in local clean PS installation, I found it and I installed but I've got this error:

![error comment module](https://user-images.githubusercontent.com/28685316/73757307-561b7680-4769-11ea-9c6d-fcecd0476cee.png)

I'm waiting your kind response.
Thanks.

**Comment by khouloudbelguith** [comment]
@ballashop, I tried with fresh install PS1.7.6.3 & it is OK.
[module_productcomments.webm.zip](https://github.com/PrestaShop/PrestaShop/files/4154232/module_productcomments.webm.zip)
Thanks!

**Comment by ballashop** [comment]
@khouloudbelguith 
I have done exactly the same steps! with a different result.
Now I will do this:
I delete the productcomments folder
I clean the cache
I close and restart MAMP
I will go through the same steps again
Then I report you.
Thanks a lot for your support.

**Comment by khouloudbelguith** [comment]
@ballashop, waiting for your feedback.

Thanks!

**Comment by ballashop** [comment]
@khouloudbelguith 
ok, in local site I installed with success, now I'm going to do the same steps with online site (into a sub folder) not the operative one  of course :), it's still to 1.6.1.24 version
I'll let you know.
Thanks for your support.

**Comment by khouloudbelguith** [comment]
@ballashop, waiting for your feedback.

Thanks!

**Comment by ballashop** [comment]
@khouloudbelguith 
amazing, online I can't find this module, I don't understand why!!!
I tried with:
comment, commenti (italian language), renew, recensioni.
I also synced to the marketplace but no term can find this module, I find it incredible.
can I copy the folder from the local server to the remote server?
Thanks for your support.

**Comment by khouloudbelguith** [comment]
@ballashop, you can search with the tech name of this module.
Try with this `productcomments`.

Thanks!

**Comment by ballashop** [comment]
I tried but it didn't work!
<img width="1093" alt="Schermata 2020-02-04 alle 16 54 16" src="https://user-images.githubusercontent.com/28685316/73761527-47d05900-476f-11ea-9274-f1f7947d0ca7.png">
<img width="1125" alt="Schermata 2020-02-04 alle 16 54 30" src="https://user-images.githubusercontent.com/28685316/73761549-4ef76700-476f-11ea-9506-e0db3d069baf.png">
<img width="1129" alt="Schermata 2020-02-04 alle 16 54 39" src="https://user-images.githubusercontent.com/28685316/73761562-528aee00-476f-11ea-8d98-54e541ab539f.png">
<img width="1135" alt="Schermata 2020-02-04 alle 16 54 47" src="https://user-images.githubusercontent.com/28685316/73761572-5585de80-476f-11ea-906e-f01e25d51f56.png">
any other suggestions?
Thanks a lot.

**Comment by khouloudbelguith** [comment]
@ballashop, what is the exact version of this shop?

Thanks!

**Comment by ballashop** [comment]
@khouloudbelguith 
1.7.6.3
Thanks

**Comment by khouloudbelguith** [comment]
@ballashop, check that the module is not installed.
Search it in the Module manager page.

Thanks!

**Comment by ballashop** [comment]
@khouloudbelguith 
it is not installed online:

<img width="1420" alt="Schermata 2020-02-04 alle 17 03 57" src="https://user-images.githubusercontent.com/28685316/73762361-65ea8900-4770-11ea-9415-1f1e47d97e2d.png">

and this is the local site:

<img width="1448" alt="Schermata 2020-02-04 alle 17 04 51" src="https://user-images.githubusercontent.com/28685316/73762448-8286c100-4770-11ea-9543-d73c6f0ea31a.png">

can I copy and paste the productcomments from local to online folder?
Thanks a lot for your support.

**Comment by khouloudbelguith** [comment]
@ballashop, try to compress the productcomments module & upload it in the second shop.

Thanks!

**Comment by ballashop** [comment]
@khouloudbelguith 
ok, I compress the productcomments folder then I uploaded by BO the zip file:

<img width="676" alt="Schermata 2020-02-04 alle 17 10 41" src="https://user-images.githubusercontent.com/28685316/73763678-0db48680-4772-11ea-9bb0-f2e672b93c2f.png">

now I have it into the online site.

<img width="1382" alt="modulo caricato" src="https://user-images.githubusercontent.com/28685316/73763704-16a55800-4772-11ea-957e-71f26fed474a.png">

The reviews are not displayed at the moment but this I think is a problem of the theme we used.
I'm setting up the theme we bought into the PS marketplace and I'll contact the developer to fix that.
The site is in debug mode and we don't have any error message!
Thanks a lot @khouloudbelguith for your valuable contribution.
